qa/rpc-tests/preciousblock.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python2
+# Copyright (c) 2015 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#
+# Test PreciousBlock code
+#
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import *
+
+class PreciousTest(BitcoinTestFramework):
+    def setup_chain(self):
+        print("Initializing test directory "+self.options.tmpdir)
+        initialize_chain_clean(self.options.tmpdir, 3)
+
+    def setup_network(self):
+        self.nodes = []
+        self.is_network_split = False 
+        self.nodes.append(start_node(0, self.options.tmpdir, ["-debug"]))
+        self.nodes.append(start_node(1, self.options.tmpdir, ["-debug"]))
+        self.nodes.append(start_node(2, self.options.tmpdir, ["-debug"]))
+
+    def run_test(self):
+        print "Mine blocks A-B-C on Node 0"
+        self.nodes[0].generate(3)
+        assert(self.nodes[0].getblockcount() == 3)
+        hashC = self.nodes[0].getbestblockhash()
+        print "Mine competing blocks E-F-G on Node 1"
+        self.nodes[1].generate(3)
+        assert(self.nodes[1].getblockcount() == 3)
+        hashG = self.nodes[1].getbestblockhash()
+        assert(hashC != hashG)
+        print "Connect nodes and check no reorg occurs"
+        connect_nodes_bi(self.nodes,0,1)
+        sync_blocks(self.nodes[0:2])
```

Comment:
FYI, i don't think this call to `sync_blocks` is doing anything helpful -- it is just checking that `getblockcount` is the same between the two nodes, which it will be immediately, even before any headers or blocks are sent between the two nodes.

Perhaps this should sync on `getchaintips()` returning the expected output?